File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
import sys
answer=0
n=int(sys.stdin.readline())
for i in range(n):
    st=[]
    str=sys.stdin.readline()
    for j in str:
        if(st and st[-1]==j):
            st.pop()
        elif(j=='A' or j=='B'):
            st.append(j)
    if(not st):
        answer+=1
print(answer)

File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include <stdio.h>
#include <iostream>
#include <string>
#include <stack>
#include <utility>
#include <algorithm>
int main(){
    std::cin.tie(0);
    std::ios::sync_with_stdio(0);
    int n{}, answer{};
    std::stack <char> st{};
    std::string str{};
    std::cin >> n;
    for(int i=0; i<n; i++){
        st={};
        std::cin >> str;
        for(char i: str){
            if(!st.empty() && st.top()==i){
                st.pop();
            }
            else{
                st.push(i);
            }
    }
                if(st.empty()){
                answer++;
            }
    }
    std::cout << answer;
}